src/App.js

```diff
@@ -1,23 +1,74 @@
-import logo from './logo.svg';
+import contactsArr from './contacts.json'
+import {useState} from 'react'
 import './App.css';
 
+
+
 function App() {
+  let fiveContacts =  [...contactsArr.slice(0, 5)]
+  
+  let [contactList, setContacts] = useState(fiveContacts)
+
+  const addContact = ( () => {
+    let random = Math.floor( (Math.random() * contactsArr.length) -1)
+    
+    setContacts( (contactList)=>{
+      const newArr = contactsArr.filter((el, index) => {
+        if (index === random){
+          return true
+        } else {
+          return false
+        }
+      })
+      return contactList.concat(newArr)
+    })
+
+  })
```

Comment:
I tried to explain that this morning, so for example like this would be a good solution:
```
const addRandomContact = ()=>{
    setContacts((prevContacts)=>{
      const newContactsList = contactsArr.filter((contact)=> {
        return !prevContacts.includes(contact);
      });
      const getRandomContact = newContactsList[Math.floor(Math.random()* newContactsList.length)];
      const newConctactSelection = [...prevContacts,getRandomContact]
      return newConctactSelection;
    })
  };
```
let me know if you have any questions